alibi/explainers/similarity/backends/pytorch/base.py

```diff
@@ -50,8 +50,26 @@ def get_grads(
         loss = loss_fn(output, Y)
         loss.backward()
         model.train(initial_model_state)
-        return np.concatenate([_PytorchBackend.to_numpy(param.grad).reshape(-1)  # type: ignore [arg-type] # see #810
-                               for param in model.parameters()])
+
+        return np.concatenate([_PytorchBackend._grad_to_numpy(grad=param.grad, name=name)
+                               for name, param in model.named_parameters()
+                               if param.grad is not None])
+
+    @staticmethod
+    def _grad_to_numpy(grad: torch.Tensor, name: Optional[str] = None) -> torch.Tensor:
```

Comment:
`grad` type should probably be a union here since it can be sparse?

Reply:
Ignore this, was thinking of the `tensorflow` backend.

Reply:
No your correct, torch also has `SparseTensors`

Reply:
Ignore me, torch handles sparse tensors by setting a layout attribute on normal tensors!